cg/meta/report/report_api.py

```diff
@@ -108,16 +133,40 @@ def render_delivery_report(self, report_data: dict) -> str:
         template = env.get_template(template_name)
         return template.render(**report_data)
 
-    def get_cases_without_delivery_report(self):
-        """Returns a list of analyses that need a delivery report"""
+    def get_cases_without_delivery_report(self, pipeline: Pipeline) -> List[models.Family]:
+        """Returns a list of cases that has been stored and need a delivery report"""
+
+        stored_cases = []
+        analyses = self.status_db.analyses_to_delivery_report(pipeline)[:50]
+
+        for analysis_obj in analyses:
+            case: models.Family = analysis_obj.family
+            last_version = self.housekeeper_api.last_version(case.internal_id)
+            hk_file = self.housekeeper_api.get_files(
+                bundle=case.internal_id, version=last_version.id if last_version else None
+            ).first()
+
+            if hk_file and Path(hk_file.full_path).is_file():
+                stored_cases.append(case)
+            else:
+                LOG.warning(
+                    f"Case {case.internal_id} must be stored before creating a delivery report"
+                )
+
+        return stored_cases
+
+    def get_cases_without_uploaded_delivery_report(self, pipeline: Pipeline) -> List[models.Family]:
+        """Returns a list of cases that need a delivery report to be uploaded"""
 
-        return self.status_db.analyses_to_delivery_report(self.analysis_api.pipeline)[:50]
+        analyses = self.status_db.analyses_to_upload_delivery_reports(pipeline)[:50]
+        cases = [analysis_obj.family for analysis_obj in analyses]
```

Comment:
Do you need the extra variable? Could you not return the list by:
`return [analysis_obj.family for analysis_obj in analyses]`